superdesk/users/async_service.py

```diff
@@ -0,0 +1,537 @@
+            raise UserNotRegisteredException("No user registered with email %s" % email_address)
+
+        return user
+
+    async def update_stage_visibility_for_users(self):
+        if not self._updating_stage_visibility:
+            return
+        logger.info("Updating Stage Visibility Started")
+        cursor = await self.find({})
+        users = await cursor.to_list()
+        for user in users:
+            await self.update_stage_visibility_for_user(user)
+
+        logger.info("Updating Stage Visibility Completed")
+
+    async def update_stage_visibility_for_user(self, user: UsersResourceModel):
+        if not self._updating_stage_visibility:
+            return
+        user_id = user.to_dict().get("_id", "")
+        try:
+            logger.info("Updating Stage Visibility for user {}.".format(user_id))
+            stages = self.get_invisible_stages_ids(user_id)
+            await self.system_update(user_id, {"invisible_stages": stages})
+            user.invisible_stages = stages
+            logger.info("Updated Stage Visibility for user {}.".format(user_id))
+        except Exception:
+            logger.exception("Failed to update the stage visibility " "for user: {}".format(user_id))
+
+    def stop_updating_stage_visibility(self):
+        if not get_app_config("SUPERDESK_TESTING"):
+            raise RuntimeError("Only allowed during testing")
+        self._updating_stage_visibility = False
+
+    def start_updating_stage_visibility(self):
+        self._updating_stage_visibility = True
+
+
+class DBUsersAsyncService(UsersAsyncService):
+    """
+    Async Service class for UsersResource and should be used when AD is inactive.
+    """
+
+    async def on_create(self, docs: list[UsersResourceModel]) -> None:
+        await super().on_create(docs)
+        for doc in docs:
+            user_dict = doc.to_dict()
+            if user_dict.get("password", None) and not is_hashed(user_dict.get("password")):
+                user_dict["password"] = get_hash(
+                    user_dict.get("password"), get_app_config("BCRYPT_GENSALT_WORK_FACTOR", 12)
+                )
+                doc = UsersResourceModel(**user_dict)
+
+    async def on_created(self, docs: list[UsersResourceModel]) -> None:
+        """Send email to user with reset password token."""
+        await super().on_created(docs)
+        resetService = get_resource_service("reset_user_password")
+        activate_ttl = get_app_config("ACTIVATE_ACCOUNT_TOKEN_TIME_TO_LIVE")
+        for doc in docs:
+            user_dict = doc.to_dict()
+            if self.user_is_waiting_activation(doc) and user_dict.get("user_type") != "external":
+                user_id = user_dict.get("_id")
+                email = user_dict.get("email")
+                username = user_dict.get("username")
+                tokenDoc = {"user": user_id, "email": email}
+                id = resetService.store_reset_password_token(tokenDoc, email, activate_ttl, user_id)
```

Comment:
`id` shadows the built-in function with the same name. It would be good not to use names that are built into the python language